cpp/src/plasma/CMakeLists.txt

```diff
@@ -126,10 +126,24 @@ if ("${COMPILER_FAMILY}" STREQUAL "gcc")
     " -Wno-conversion")
 endif()
 
+list(APPEND PLASMA_EXTERNAL_STORE_SOURCES "external_store_worker.cc" "external_store.cc" "hash_table_store.cc")
+
+if (PLASMA_EXTRA_EXTERNAL_STORE_SOURCES)
+  list(APPEND PLASMA_EXTERNAL_STORE_SOURCES ${PLASMA_EXTRA_EXTERNAL_STORE_SOURCES})
+endif ()
+
+if (PLASMA_EXTERNAL_STORE_LIBDIRS)
+  link_directories(${PLASMA_EXTERNAL_STORE_LIBDIRS})
+endif ()
+
+if (NOT APPLE AND UNIX)
```

Comment:
Please don't set `--no-as-needed` globally. Why do you need this?

Reply:
You're right, this is not necessary, I'll remove it.